packages/mdc-slider/foundation.ts

```diff
@@ -21,113 +21,123 @@
-      getAttribute: (/* name: string */) => /* string|null */ null,
-      setAttribute: (/* name: string, value: string */) => {},
-      removeAttribute: (/* name: string */) => {},
-      computeBoundingRect: () => /* ClientRect */ ({
-        top: 0, right: 0, bottom: 0, left: 0, width: 0, height: 0,
-      }),
-      getTabIndex: () => /* number */ 0,
-      registerInteractionHandler: (/* type: string, handler: EventListener */) => {},
-      deregisterInteractionHandler: (/* type: string, handler: EventListener */) => {},
-      registerThumbContainerInteractionHandler: (/* type: string, handler: EventListener */) => {},
-      deregisterThumbContainerInteractionHandler: (/* type: string, handler: EventListener */) => {},
-      registerBodyInteractionHandler: (/* type: string, handler: EventListener */) => {},
-      deregisterBodyInteractionHandler: (/* type: string, handler: EventListener */) => {},
-      registerResizeHandler: (/* handler: EventListener */) => {},
-      deregisterResizeHandler: (/* handler: EventListener */) => {},
-      notifyInput: () => {},
-      notifyChange: () => {},
-      setThumbContainerStyleProperty: (/* propertyName: string, value: string */) => {},
-      setTrackStyleProperty: (/* propertyName: string, value: string */) => {},
-      setMarkerValue: (/* value: number */) => {},
-      appendTrackMarkers: (/* numMarkers: number */) => {},
-      removeTrackMarkers: () => {},
-      setLastTrackMarkersStyleProperty: (/* propertyName: string, value: string */) => {},
-      isRTL: () => /* boolean */ false,
-    });
+  static get defaultAdapter(): MDCSliderAdapter {
+    // tslint:disable:object-literal-sort-keys
+    return {
+      hasClass: () => false,
+      addClass: () => undefined,
+      removeClass: () => undefined,
+      getAttribute: () => null,
+      setAttribute: () => undefined,
+      removeAttribute: () => undefined,
+      computeBoundingRect: () => ({top: 0, right: 0, bottom: 0, left: 0, width: 0, height: 0}),
+      getTabIndex: () => 0,
+      registerInteractionHandler: () => undefined,
+      deregisterInteractionHandler: () => undefined,
+      registerThumbContainerInteractionHandler: () => undefined,
+      deregisterThumbContainerInteractionHandler: () => undefined,
+      registerBodyInteractionHandler: () => undefined,
+      deregisterBodyInteractionHandler: () => undefined,
+      registerResizeHandler: () => undefined,
+      deregisterResizeHandler: () => undefined,
+      notifyInput: () => undefined,
+      notifyChange: () => undefined,
+      setThumbContainerStyleProperty: () => undefined,
+      setTrackStyleProperty: () => undefined,
+      setMarkerValue: () => undefined,
+      appendTrackMarkers: () => undefined,
+      removeTrackMarkers: () => undefined,
+      setLastTrackMarkersStyleProperty: () => undefined,
+      isRTL: () => false,
+    };
+    // tslint:enable:object-literal-sort-keys
   }
 
+  private rect_: ClientRect | null = null;
```

Comment:
should this just be `this.rect_!: ClientRect;`? You're using the `!` below, meaning it should never be null.

Reply:
Done